lib/addons/opa-gatekeeper/index.ts

```diff
@@ -0,0 +1,85 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { setPath } from "../../utils/object-utils";
+
+
+/**
+ * Properties available to configure opa gatekeeper
+ */
+export interface OpaGatekeeperAddOnProps {
+    /**
+     * Create namespace for Gatekeeper controller 
+     * @default true
+     */
+    createNamespace?: boolean;
+
+    /** Default namespace
+     * @default gatekeeper-system
+     */
+    namespace?: string;
+
+    /**
+     * Helm chart version to use to install.
+     * @default 3.7.0-beta.1
+     */
+    version?: string;
+
+    /**
+     * Setting validatingWebhookFailurePolicy to True
+     * @default true 
+     */
+    disableValidatingWebhook?: boolean;
+
+    /**
+     * Label namespace so that webhook doesn't interfere with deployment of gatekeeper chart
+     * @default true
+     */
+    labelNamespace?: boolean;
+
+    /**
+     * values to pass to the helm chart per https://github.com/open-policy-agent/gatekeeper/tree/master/charts/gatekeeper
+     */
+     values?: {
+        [key: string]: any;
+    };
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: OpaGatekeeperAddOnProps = {
+    namespace: 'gatekeeper-system',
+    version: '3.7.0-beta.1',
+    disableValidatingWebhook: false,
+    labelNamespace: true
+};
+
+
+export class OpaGatekeeperAddOn implements ClusterAddOn {
+    
+    readonly options: OpaGatekeeperAddOnProps;
+    
+    constructor(props?: OpaGatekeeperAddOnProps) {
+        this.options = { ...defaultProps, ...props };
+    }
+
+
+    deploy(clusterInfo: ClusterInfo): void {
+
+        const props = this.options;
+
+        const values = { ...props.values ?? {}};
+
+        setPath(values, 'disableValidatingWebhook', false)
```

Comment:
You are hardcoding values for disableValidationWebhook and labelNamespace, basically means that *regardless* of what customer passes, you will set them to these values. This is wrong, you should propagate values that you allow customers to pass. 

Imagine customer is using your add-on like this:

```typescript
new OpaGateKeeperAddOn({
   labelNamespace: false,
   disableValidationWebhook: true
);
```

The above will have no effect.